pleiades/vocabularies/browser/vocab.py

```diff
@@ -122,3 +123,22 @@ def publishTraverse(self, request, name):
             return VocabView(self.context, request, name)
         return super(PleiadesVocabularyPublishTraverse, self).publishTraverse(
             request, name)
+
+
+class SearchUtilities(BrowserView):
+
+    def __init__(self, context, request):
+        super(SearchUtilities, self).__init__(context, request)
+        self.catalog = getToolByName(context, 'portal_catalog')
+
+    def get_place_type_data(self):
+        featureTypes = set(self.catalog.uniqueValuesFor('getFeatureType'))
+        places_in_use = sorted(featureTypes)
+        place_types = get_vocabulary('place_types')
+        data = []
+        for pl in places_in_use:
+            if pl != '':
+                entry = [{'title': x['title'], 'id':x['id']} \
+                    for x in place_types if x['id'] == pl]
```

Comment:
Rather than looping over the vocabulary for every place in use, why not convert place_types into a dict and do a fast lookup. That way this code runs in O(2*n) rather than O(n^2):

    places = {p['id']: p['title'] for p in place_types}
    data = [{'id': p, 'title': places[p]} for p in places_in_use if p and p in places]